enhancements/installer/aws-load-balancer-type.md

```diff
@@ -0,0 +1,343 @@
+type AWSIngressSpec struct {
+  // type allows user to set a load balancer type.
+  // When this field is set the default ingresscontroller will get created using the specified LBType.
+  // If this field is not set then the default ingress controller of LBType Classic will be created.
+  // Valid values are: 
+  // * "Classic": A Classic Load Balancer that makes routing decisions at either
+  //   the transport layer (TCP/SSL) or the application layer (HTTP/HTTPS). See
+  //   the following for additional details:
+  //   https://docs.aws.amazon.com/AmazonECS/latest/developerguide/load-balancer-types.html#clb
+  //
+  // * "NLB": A Network Load Balancer that makes routing decisions at the
+  //   transport layer (TCP/SSL). See the following for additional details:
+  //   https://docs.aws.amazon.com/AmazonECS/latest/developerguide/load-balancer-types.html#nlb
+  // +unionDiscriminator
+  // +kubebuilder:validation:Enum:=NLB;Classic
+  // +kubebuilder:validation:Required 
+  Type AWSLBType `json:"type,omitempty"`
+}
+
+type AWSLBType string
+
+const (
+    // NLB is the Network Load Balancer Type of AWS. Using NLB one can set NLB load balancer type for the default ingress controller.
+    NLB AWSLBType = "NLB"
+
+    // Classic is the Classic Load Balancer Type of AWS. Using CLassic one can set Classic load balancer type for the default ingress controller.
+    Classic AWSLBType = "Classic"
+)
+```
+
+#### Resource provided to the installer
+
+The users provide a load balancer type to the installer.
+
+```yaml
+apiVersion: v1
+baseDomain: example.com
+metadata:
+  name: test-cluster
+platform:
+  aws:
+    region: us-west-2
+    lbType: nlb
+    subnets:
+    - subnet-1
+    - subnet-2
+    - subnet-3
+pullSecret: '{"auths": ...}'
+sshKey: ssh-ed25519 AAAA...
+```
+
+
+#### Resources created by the installer
+
+- The installer will create infrastructure cluster object with the load balancer type defined in it.
+
+```yaml
+ oc edit ingress.config/cluster -o yaml
+ # Please edit the object below. Lines beginning with a '#' will be ignored,
+ # and an empty file will abort the edit. If an error occurs while saving this file will be
+ # reopened with the relevant failures.
+ #
+ apiVersion: config.openshift.io/v1
+ kind: Ingress
+ metadata:
+   creationTimestamp: "2022-08-30T01:03:17Z"
+   generation: 11
+   name: cluster
+   resourceVersion: "60888"
+   uid: de89cc95-449e-4ae5-a837-f4171fe61e36
+ spec:
+   domain: apps.testnlbapi.devcluster.openshift.com
+   loadbalancer:
+     platform:
+       aws:
+         type: NLB
+       type: AWS
+ status:
+   componentRoutes:
+```
+
+- The cluster-ingress-operator will check the cluster ingress object for the lbType and reconcile to create an default ingress controller
+  with the specified lbType.
+
+#### Destroying cluster
+
+- None
+
+
+### Risks and Mitigations
+
+- Users need to put either nlb or elb as input or else the LB creation will fail.
+
+- To mitigate this we need to validate the input.
```

Comment:
```suggestion
- Users need to put either "NLB" or "Classic" as input or else the LB creation will fail.

- To mitigate this we need to validate the input.
```
This might be a good place to mention that there are known issues with using NLBs, such as <https://bugzilla.redhat.com/show_bug.cgi?id=2023681>.

Reply:
I suggested documenting disadvantages to using NLBs, such as <https://bugzilla.redhat.com/show_bug.cgi?id=2023681> in order to forestall questions like this: <https://github.com/openshift/installer/pull/6478#discussion_r992741999>.  Any reason not to document this?